Microsoft.Research/ManagedContract.Setup/DotNet.Contracts.targets

```diff
@@ -0,0 +1,14 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <CodeContractsInstallDir>$(MSBuildThisFileDirectory)</CodeContractsInstallDir>
```

Comment:
That's backwards from how the package would typically be used.

If you don't have CC installed, the package should use it's own version.

If you do have CC installed and no package, you would use the system version.

If you have CC installed and you have projects that use different package versions, each project should use the version that it requires, or was last tested with, or whatever. Modifying the system version should introduce changes to a project that has taken a dependency on a _specific version_ of the NuGet package.

I see absolutely no point in having system Code Contracts installed, then installing a NuGet package, and having the package just defer to the global installation that was already there and being used. The package becomes redundant and confusing.